src/components/PinInput/src/PinInput.vue

```diff
@@ -0,0 +1,308 @@
+<template>
+	<div>
+		<div
+			:style="{ width: containerWidth }"
+			:class="{
+				[$s.PinInputContainer]: true,
+				[$s.shake]: isShaking,
+				[$s.disabled]: disabled,
+			}"
+		>
+			<div
+				v-for="(val, i) in code"
+				:key="i"
+				:class="$s.CodeInputWrapper"
+			>
+				<input
+					:ref="getCodeCellRef(i)"
+					:value="val"
+					:autocomplete="i === 0 ? 'one-time-code' : 'off'"
+					:disabled="disabled"
+					:maxlength="i === 0 ? pinLength : 1"
+					:class="{
+						[$s.CodeInput]: true,
+						[$s.filled]: variant === 'fill',
+						[$s.error]: invalid,
+					}"
+					type="text"
+					inputmode="numeric"
+					pattern="[0-9]*"
```

Comment:
is this i18n friendly? it's okay if the answer is no, but just wanna make sure we're only supporting PINs that are single-digit numbers only and that we're not shipping this to any countries that have PINs with weird stuff like letters or whatever in them. (yes i know the N in PIN stands for Number so common sense would tell us PINs only consist of numbers but I just wanna be extra careful and double-check).

Reply:
The pins that our services send are always numbers so I want to say that this is i18n friendly yeah